tests/integration/test_storage_delta/test.py

```diff
@@ -157,6 +179,82 @@ def create_delta_table(node, table_name, bucket="root"):
     )
 
 
+def get_creation_expression(
+    storage_type,
+    table_name,
+    cluster,
+    format="Parquet",
+    table_function=False,
+    allow_dynamic_metadata_for_data_lakes=False,
+    run_on_cluster=False,
+    **kwargs,
+):
+    allow_dynamic_metadata_for_datalakes_suffix = (
+        " SETTINGS allow_dynamic_metadata_for_data_lakes = 1"
+        if allow_dynamic_metadata_for_data_lakes
+        else ""
+    )
+
+    if storage_type == "s3":
+        if "bucket" in kwargs:
+            bucket = kwargs["bucket"]
+        else:
+            bucket = cluster.minio_bucket
+
+        if run_on_cluster:
+            assert table_function
+            return f"deltalakeS3Cluster('cluster_simple', s3, filename = '{table_name}/', format={format}, url = 'http://minio1:9001/{bucket}/')"
+        else:
+            if table_function:
+                return f"deltalakeS3(s3, filename = '{table_name}/', format={format}, url = 'http://minio1:9001/{bucket}/')"
+            else:
+                return (
+                    f"""
+                    DROP TABLE IF EXISTS {table_name};
+                    CREATE TABLE {table_name}
+                    ENGINE=DeltaLake(s3, filename = '{table_name}/', format={format}, url = 'http://minio1:9001/{bucket}/')"""
+                    + allow_dynamic_metadata_for_datalakes_suffix
+                )
+
+    elif storage_type == "azure":
+        if run_on_cluster:
+            assert table_function
+            return f"""
+                deltalakeAzureCluster('cluster_simple', azure, container = '{cluster.azure_container_name}', storage_account_url = '{cluster.env_variables["AZURITE_STORAGE_ACCOUNT_URL"]}', blob_path = '/{table_name}', format={format})
+            """
+        else:
+            if table_function:
+                return f"""
+                    deltalakeAzure(azure, container = '{cluster.azure_container_name}', storage_account_url = '{cluster.env_variables["AZURITE_STORAGE_ACCOUNT_URL"]}', blob_path = '/{table_name}', format={format})
```

Comment:
It seems this test would not work, because I do not see any change in `src/TableFunctions` in this PR, but there should be some adjustments, see https://github.com/ClickHouse/ClickHouse/blob/720ace40ebb50987ccf41f9de16d6a865470b2c0/src/TableFunctions/TableFunctionObjectStorage.cpp#L241-L243

Reply:
ya right, missed it. Addressed it